src/sign-module.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+echo "Running the Axcient Kernel Module signer"
+
+result=$(mokutil --sb-state)
+if [[ "$result" != *"enabled"* ]]; then
+    echo "Detected a BIOS firmware or the Secure Boot feature is disabled"
+    exit 1
+else 
+    keys=$(cat /proc/keys | grep "asymmetri Axcient")
+    if [ -n "$keys" ]; then 
+	echo "Starting the setup"
+    else
+	echo "WARNING: Secure Boot detected on this system. Either disable EFI Secure Boot "
+	"or check with Axcient knowledgebase for instructions "
+	"for enrolling a signing key for secure DKMS module support." >&2 \
+	&& exit 0
+    fi	
+fi
+
+echo "Kernel Version: ${1}"
+echo "Module Path: ${2}"
+
+hash_algo=sha256
+private_key=/root/axcient-driver.key
+x509_cert=/root/axcient-driver.der
+
+if [  -n "$(uname -a | grep Ubuntu)" ]; then
+       prefix=/usr/src/linux-headers-
+else
+       prefix=/usr/src/kernels/
+fi
```

Comment:
More safety check destination file instead of OS
```
sign_util=/usr/src/linux-headers-${1}/scripts/sign-file 
if [ ! -e "${sign_util}" ]; then
   sign_util=/usr/src/kernels//${1}/scripts/sign-file 
fi
if [ ! -e "${sign_util}" ]; then
  echo "WARNING! cannot find sign tool"
  exit 0
fi
```